### PR TITLE
Add nameable decorator classes

### PR DESCRIPTION
**Pull request by dennis-every**
In this pull request, we completed the following:
- Add nameable decorator classes

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    puts 'NotImplementedError'
+  end
```
Comment (christianbiring1) [inline nameable.rb]: - [ ] Kindly make sure your `correct_name` method rise an error properly, you weren't asked to display it as a text the way you did here,

Hint: To raise an error in ruby, You may consider saying:
```rb
def correct_name
        raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
    end
```

File: person.rb [hunk 2]
```diff
@@ -1,8 +1,11 @@
-class Person
+require './nameable'
+
+class Person < Nameable
   attr_reader :id
   attr_accessor :name, :age
 
   def initialize(age, name = 'Unknown', parent_permission: true)
```
Comment (christianbiring1) [inline person.rb:7]: - [ ] Kindly make sure the user can also have the possibility to change the parent permission apart from the default one for a better user experience.
You may achieve this by saying: 
```suggestion
  def initialize(age, name = 'Unknown', parent_permission = true)
```
And also in the `can_use_services?` make sure you only have
```rb
of_age? || @parent_permission
```

File: base_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable
```
Comment (christianbiring1) [inline base_decorator.rb]: - [ ] As specified in the project requirement, the `BaseDecorator` is supposed to take the class `Neamable` as param and return the correct name like so:

```rb
def correct_name
      @nameable.correct_name
    end
```

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @dennis-every , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
# STATUS: REQUEST CHANGES :recycle: 
soclose

## Highlights
- [ ] All linters checks are green :heavy_check_mark: 
- [ ] Descriptive PR `title` and `summary` :+1: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (dennis-every) [reply person.rb:7]: > * [ ]  Kindly make sure the user can also have the possibility to change the parent permission apart from the default one for a better user experience.
>   You may achieve this by saying:
> 
> And also in the `can_use_services?` make sure you only have
> 
> ```ruby
> of_age? || @parent_permission
> ```

Hi @christianbiring1  - with the existing code the user can change the parent permission by passing a keyword argument e.g. parent_permission: false or parent_permission: true

Comment (dennis-every) [reply person.rb:7]: @christianbiring1 - additionally it seems it's also recommended by rubocop, see below:
Screenshot 2023-04-12 at 1 35 38 PM

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @dennis-every 👋, 

Great work on making all the changes requested by the previous reviewer 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**